lib/kitchen/driver/aws/instance_generator.rb

```diff
@@ -134,14 +143,15 @@ def debug_if_root_device(bdms)
 
         def prepared_user_data
           # If user_data is a file reference, lets read it as such
-          unless @user_data
-            if config[:user_data] && File.file?(config[:user_data])
+          return nil if config[:user_data].nil?
+          @user_data ||= begin
+            if File.file?(config[:user_data])
               @user_data = File.read(config[:user_data])
             else
               @user_data = config[:user_data]
             end
+            @user_data = Base64.encode64(@user_data)
```

Comment:
Interesting, that seems to be a low-level concern that we need to deal with as a user of the sdk. Ah well